### PR TITLE
adds change detection performance enhancements

### PR DESCRIPTION
**Pull request by esotrope**

### CODE REVIEW

File: webapp/src/main/webapp/src/app/aggregate-report/aggregate-report-summary.component.ts [hunk 1]
```diff
@@ -35,7 +35,8 @@ const WideColumnProvider: ColumnProvider = createColumnProvider();
   templateUrl: 'aggregate-report-summary.component.html',
   host: {
     'class': 'aggregate-report-summary'
-  }
+  },
+  changeDetection: ChangeDetectionStrategy.OnPush
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/aggregate-report/aggregate-report-summary.component.ts]: this actually made a noticeable difference in the render time for both the summary and table views. 
We are able to apply this change detection strategy to views that are "readonly" or "input-only" other views that are readonly and "read-once" at init time can be optimized by calling `changeDetector.detach()`

File: webapp/src/main/webapp/src/app/aggregate-report/query-forms/claim-report-form.component.html [hunk 2]
```diff
@@ -442,7 +442,7 @@ <h3>{{'aggregate-report-form.section.subgroup-filters-heading' | translate}}</h3
     <div class="well-group">
       <div class="well">
         <h3>{{'aggregate-report-form.section.review-heading' | translate}}</h3>
-        <ng-container *ngIf="formGroup.valid">
+        <ng-container *ngIf="formGroup.valid && capableOfRowEstimation()">
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/aggregate-report/query-forms/claim-report-form.component.html]: this was in the wrong place in a few of the templates. an old copy-paste error probably

File: webapp/src/main/webapp/src/app/aggregate-report/query-forms/claim-report-form.component.html [hunk 3]
```diff
@@ -593,3 +593,10 @@ <h3>{{'aggregate-report-form.section.preview-heading' | translate}}</h3>
           (click)="submitAction.emit($event)">{{'aggregate-report-form.submit' | translate}}
   </button>
 </ng-template>
+
+<ng-template #warning let-text>
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/aggregate-report/query-forms/claim-report-form.component.html]: another old copy paste error i think - this template was being referenced in these views but was not present

File: webapp/src/main/webapp/src/app/aggregate-report/results/aggregate-report-table.component.html [hunk 4]
```diff
@@ -170,8 +170,7 @@
 
           <div *ngSwitchCase="'studentsTested'"
                class="text-right">
-            <!--{{rowData.studentsTested}}-->
-            {{rowData.itemId}}
+            {{rowData.studentsTested}}
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/aggregate-report/results/aggregate-report-table.component.html]: going to merge this PR since this was a bug introduced by my other PR

File: webapp/src/main/webapp/src/app/aggregate-report/results/aggregate-report-table.component.ts [hunk 5]
```diff
@@ -595,12 +596,13 @@ export class AggregateReportTableComponent implements OnInit {
...
       if (_.isEqual(previousColumns.slice().sort(), newColumns.slice().sort())) {
-        this.columns = sortColumns(
+        const sortedColumns = sortColumns(
           columns,
           newColumns
         );
+        this.columns = sortedColumns;
         this._identityColumnComparators = identityColumnComparators(
-          columns,
+          sortedColumns,
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/aggregate-report/results/aggregate-report-table.component.ts]: fixes a column sorting bug

File: webapp/src/main/webapp/src/app/aggregate-report/query-forms/claim-report-form.component.html (hunk 3 above)
Comment (malaffoon) [reply webapp/src/main/webapp/src/app/aggregate-report/query-forms/claim-report-form.component.html]: I know these are html templates so i probably won't like the answer but ... is there any sort of test we can put in place to catch stuff like this. A missing template seems like a testable thing.

Comment (esotrope) [reply webapp/src/main/webapp/src/app/aggregate-report/query-forms/claim-report-form.component.html]: you're right it is. i'll see if i cant add something to cover it next go around
